equational_theories/ParseImplications.lean

```diff
@@ -12,6 +12,15 @@ structure Implication where
   rhs : String
 deriving Lean.ToJson, Lean.FromJson
 
+/--
+Names of equations satisfied rsp. refuted by the same magma; used to concisely express
+antiimplications.
```

Comment:
Either here or in `FactsSyntax.lean` (probably there is better) I'd like to see it explicitly worked out which negated implications a `Facts` entails. (I, for one, have not internalized De Morgan's laws deeply enough for this to be immediately obvious to me.)

For example, if I understand correctly,
```
Facts G [1, 2] [4, 5]
```
entails `not (Equation4 -> Equation1)` and `not (Equation4 -> Equation2)` and `not (Equation5 -> Equation1)` and `not (Equation5 -> Equation2).`

Reply:
Almost: It proofs that these implications don’t hold: 1→4, 1→5, 2→4, 2→5. The first list is the list of equations that are satisfied.

Reply:
```suggestion
antiimplications.

For example, if `satisfied = [1, 2]` and `refuted = [4, 5]`, then this expresses
the four antiimplications `¬ 1→4`, `¬ 1→5`, `¬ 2→4`, `¬ 2→5`.
```

Reply:
Thanks. My current code on `main` gets this backwards! (Though maybe any downstream consumers re-flip it.)